weave-js/src/components/Panel2/ChildPanelExportReport/ChildPanelExportReport.tsx

```diff
@@ -44,6 +45,25 @@ export const ChildPanelExportReport = ({
   const selectedPath = useSelectedPath();
   const closeDrawer = useCloseDrawer();
 
+  // Export is only allowed for panels *inside* main (excluding main itself)
+  useEffect(() => {
+    if (selectedPath[0] !== 'main' || selectedPath.length < 2) {
```

Comment:
nit: can this condition be put into a function. The intended outcome is that you only want top-level panels from the main section of a dashboard to be exportable. This can be encapsulated into a function. The reason i ask is that `selectedPath[0] !== 'main' || selectedPath.length < 2` is "correct", yet "fragile"... since it relies on dashboards being defined with a top-level `main` section which could change in the future.

Reply:
Good point. I'll refactor this separately since I see similar logic in [a few places](https://github.com/search?q=repo%3Awandb%2Fweave+%22%3D%3D%3D+%27main%27%22&type=code) and it would be nice to address all together